control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -211,33 +173,120 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
-	oldStatus := hostedControlPlane.Status.DeepCopy()
+	// Always try and update status if it has changed
+	// TODO(dmace): alternatively, refactor to always recalculate status up front
+	defer func() {
+		if err := r.Client.Status().Update(ctx, hostedControlPlane); err != nil {
+			r.Log.Error(err, "failed to update status", "hostedcontrolplane", req.String())
+		}
```

Comment:
I don't like that we wouldn't return an error if we fail to update status. Can we break this function up at this spot into another function that takes a HostedControlPlane (something like `func reconcileHostedControlPlane(ctx context.Context, hcp *hyperv1.HostedControlPlane) (ctrl.Result, error)`) and when we get back from that other function try to update the status?

Reply:
That sounds better than what I have here, I'll try the refactor and see how it feels...

Reply:
Still haven't done any testing, but I took your advice and continued the refactor to the point where now all status is computed up front, and the control plane update happens in its own discrete step after status. See what you think...

Reply:
Looks good to me